### PR TITLE
Defer flag

### PR DESCRIPTION
**Pull request by joelbirchler**
Added a --defer flag that defers automatic execution of jasmine. This is necessary for systems like RequireJS (AMD) that depend on async loading.

### CODE REVIEW

**Comment by rlayte** [comment]
Can you add an example of how this will work so it can be tested?

**Comment by joelbirchler** [comment]
I can write a test for this, if you'd like, but I'm not sure how we want to get around the single config.json.

An application using RequireJS might have 2 modules like this:

_src/main.js_

```
define(['./submodule'],function(submodule){
  return {
    init: function(){
      console.log('Hello from main.js!');
      submodule.init();
    }
  };
});
```

_src/submodule.js_

```
define([], function(){
  return {
    init: function(){
      console.log('Hello from submodule.js!');
    }
  };
});
```

The _spec/config.json_ only includes RequireJS:

```
["lib/require.js"]
```

And we write our specs in modules too. Here is one:

```
define(['src/main'], function(main) {

  describe("main should be sane", function() {

    it("should have tests", function() {
      expect(true).toBe(true);
    });

  });  

});
```

Then we have a _spec/runner.js_ which requires the specs and then kicks off Jasmine. RequireJS will automatically identify and load dependencies down the tree.

```
require(['spec/main-spec', 'spec/submodule-spec'], function(main, submodule) {
  jasmine.getEnv().execute();  
});
```

We could have written this in a way that would automatically populate the specs in the runner.js, but we wanted to keep the tight-coupling to RequireJS out of Aladdin. The addition of --defer adds a generic way to solve our problem and adds some flexibility.

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "tdd",
     "headless"
   ],
-  "author": "Richard Layte",
+  "author": "Joel Birchler",
```
Comment (joelbirchler) [inline package.json]: Woops! I will remove this.

**Comment by joelbirchler** [comment]
I updated the commit a bit and squashed it. Sorry about the author screw up, I didn't mean for that to go in the PR.

**Comment by rlayte** [comment]
Sorry for the delay in merging this.

I'm still not 100% convinced by this because it breaks the ability to run specific specs from the cli. Would be interested to hear if this has been an issue for you.

**Comment by joelbirchler** [comment]
Thanks for the merge!

You make a good point. Running a specific spec hasn't been an issue for us because we are building many separate small apps. It's not clear to me how to support both asynchronous loading and individual spec testing without building something specific to RequireJS. If we baked in support for specifically RequireJS, the runner.js could be added to the runner.html with a dynamic listing of specs to load. I went for the generic --defer approach because I thought it might appeal to more users.

**Comment by rufenerd** [comment]
Thanks!

**Pull request merged**